DOCS-3158: document updated table widget, and also all other viz widgets (#4420)
viamrobotics/docs: docs/data-ai/data/visualize.md

### Prerequisites

> {{% expand "A machine with at least one sensor" %}}

Author: Thanks, great call!

> Make sure your machine has at least one configured sensor component.

Reviewer: Looks like the expander heading was updated per Naomi's comment, but the expander contents should match:
```suggestion
Make sure your machine is configured with at least one component that can capture data, for example:

- [Sensor](/operate/reference/components/sensor/)
- [Movement sensor](/operate/reference/components/movement-sensor/)
- [Camera](/operate/reference/components/camera/)
```

{{% /expand%}}

### Create a workspace

1. Navigate to the **FLEET** page's [**TELEOP** tab](https://app.viam.com/teleop).
   Click **+ Create workspace**.
   Enter a unique name for your workspace in the top left of the page, replacing the placeholder `example-workplace` text.

1. From the **Select location** dropdown, select the location that contains the machine that you would like to visualize data from.

1. From the **Select machine** dropdown, select the machine that you would like to visualize data from.

### Add a widget

1. Click **Add widget** and select a widget type to create a new widget of that type on your workspace:

1. To configure the widget, click the pencil icon in the top right of your widget:

   {{<imgproc src="/services/data/visualize-widget-configure.png" alt="Click the pencil icon to configure your widget." style="width:500px" resize="1200x" class="imgzoom fill shadow" >}}

   {{% alert title="Tip" color="tip" %}}
   Select a location and machine for your workspace to populate the widget configuration dropdowns using sensor names and data from that machine.
   {{% /alert %}}

> 1. To arrange widgets on your workspace, click and drag the grid icon in the top left of your widget:

Author: Since you can arrange even a single widget, I think it's fine here.

> 1. To arrange widgets on your workspace, click and drag the grid icon in the top left of your widget:

Reviewer: I tried doing it and I couldn't get any changes to happen--it still takes up the whole screen. What does it mean to arrange a single widget?

### Widget types

Viam provides the following types of widgets that you can customize to visualize data synced from your machines:

#### Camera stream

The camera stream widget displays a live feed of the most recent image captured by a camera component:

{{<imgproc src="/services/data/visualize-widget-camera.png" resize="800x" style="width: 500px" class="fill imgzoom shadow" declaredimensions=true alt="A camera widget displaying a live camera feed.">}}

#### GPS

The GPS widget displays the current GPS location of any sensor that reports a position:

{{<imgproc src="/services/data/visualize-widget-gps.png" resize="800x" style="width: 500px" class="fill imgzoom shadow" declaredimensions=true alt="A GPS widget displaying a live location.">}}

#### Stat

The stat widget displays the most recent reading recorded by any sensor that produces tabular data:

{{<imgproc src="/services/data/visualize-widget-stat.png" resize="800x" style="width: 500px" class="fill imgzoom shadow" declaredimensions=true alt="A stat widget displaying a live sensor reading.">}}

> To configure the stat widget, choose a sensor from the **Sensor name** dropdown. Then, select the reading you would like to display from the **Path** dropdown. Finally, assign a title, a unit suffix, and a refresh rate.

Author: ```suggestion
To configure the stat widget:

1. Choose a sensor from the **Sensor name** dropdown.
1. Select the reading you would like to display from the **Path** dropdown.
1. Assign a title, a unit suffix, and a refresh rate.
```

### Prerequisites

{{% expand "A machine with at least one sensor, camera, or other component that captures data" %}}

Make sure your machine is configured with at least one component that can capture data, for example:

- [Sensor](/operate/reference/components/sensor/)
- [Movement sensor](/operate/reference/components/movement-sensor/)
- [Camera](/operate/reference/components/camera/)

> For a full list of supported sensors, see [sensor components](/operate/reference/components/sensor/).
> 

Reviewer: ```suggestion
```

To configure the time series widget, define the following attributes for each line in the time series:

1. From the **Resource name** dropdown, choose a sensor you would like to visualize.
1. From the **Capture method** dropdown, choose a method of data capture (for example **Readings**).
1. From the **Path** dropdown, choose the field of data that this line should visualize.

Use the other fields to customize the title, unit, duration, and other aspects of your visualization.

The **window method** allows you to aggregate sensor readings over specified time intervals instead of displaying raw data points.
Select a window method from the following options:

- **None**: shows raw data with the path specified with no aggregation
- **Count**: shows the number of readings within the window
- **Average**: calculates the average value throughout the window
- **Minimum**: shows the minimum value within the window
- **Maximum**: shows the maximum value within the window
- **Custom**: shows the result of a custom MQL aggregation pipeline that you define

#### Table

> The table widget displays a grid of historic tabular data values. You can display multiple fields simultaneously in a single table. Each row in the table represents a separate historic reading; each column represents a field.

Reviewer: ```suggestion
The table widget displays a grid of historic tabular data values. You can display multiple fields simultaneously in a single table.
Each row in the table represents a separate historic reading; each column represents a field.
```
[nit]

{{<imgproc src="/services/data/visualize-widget-table.png" resize="800x" style="width: 500px" class="fill imgzoom shadow" declaredimensions=true alt="A table widget displaying a grid of sensor readings.">}}

To configure the table widget, define the following attributes:

1. From the **Resource name** dropdown, choose a sensor you would like to visualize.
1. From the **Capture method** dropdown, choose a method of data capture (for example **Readings**).
1. From the **Path** dropdown, choose the data that this table should visualize.

> Use a custom MQL aggregation pipeline stage (or series of stages) to transform your sensor data into a flat object where each field corresponds to a column in the table. Consider the following sensor data, which contains information about air quality in a field named `readings`:

Reviewer: ```suggestion
Use a custom MQL aggregation pipeline stage (or series of stages) to transform your sensor data into a flat object where each field corresponds to a column in the table.
Consider the following sensor data, which contains information about air quality in a field named `readings`:
```
[nit]

```json
"data" {
 "readings": {
  "gas_resistance": 114978.66606781945,
  "temperature": 22.96,
  "pressure": 1016.18,
  "humidity": 48.318
}
```

> To visualize this data in a table with columns titled "gas_resistance", "temperature", "pressure", and "humidity", use the following custom MQL stage:

Reviewer: ```suggestion
To visualize this data in a table with columns titled "gas_resistance," "temperature," "pressure," and "humidity," use the following custom MQL stage:
```
[nit]
This would be different if you were using monospace (which I think you could here?) but if not, normal punctuation convention applies

> 

Reviewer: out-of-diff: noticed the prereq says at least one of the following and then has a single bullet. consider breaking sensor and movement sensor into separate bullets, and linking each to its respective component page instead of the configure page.

{{% /expand%}}

### Create a workspace

1. Navigate to the **FLEET** page's [**TELEOP** tab](https://app.viam.com/teleop).
   Click **+ Create workspace**.
   Enter a unique name for your workspace in the top left of the page, replacing the placeholder `example-workplace` text.

>    {{<imgproc src="/services/data/visualize-choose-workspace.png" resize="800x" style="width: 700px" class="fill imgzoom shadow" declaredimensions=true alt="On the TELEOP page, you can choose an existing workspace or create a new workspace.">}}

Reviewer: All 3 of these screenshots feel unnecessary

1. From the **Select location** dropdown, select the location that contains the machine that you would like to visualize data from.

   {{<imgproc src="/services/data/visualize-choose-location.png" resize="800x" style="width: 700px" class="fill imgzoom shadow" declaredimensions=true alt="Choosing a location for your workspace.">}}

1. From the **Select machine** dropdown, select the machine that you would like to visualize data from.

   {{<imgproc src="/services/data/visualize-choose-machine.png" resize="800x" style="width: 700px" class="fill imgzoom shadow" declaredimensions=true alt="Choosing a machine for your workspace.">}}

### Add a widget

1. Click **Add widget** and select a widget type to create a new widget of that type on your workspace:

>    {{<imgproc src="/services/data/visualize-add-widget.png" resize="800x" style="width: 300px" class="fill imgzoom shadow" declaredimensions=true alt="Choosing a widget type from the widget menu.">}}

Reviewer: the first 3 screenshots in this section also seem unnecessary since the UI is pretty clear

Author: I disagree completely. The widget configuration and movement buttons have no text, no tooltip, and confused me on first use. Removed the first screenshot since that UI is clear enough.

Reviewer: Hmm maybe where it says "Use the widget header to configure" suggest to app that they make it "Use the edit button to configure" or generally change the UI?

1. To configure the widget, click the pencil icon in the top right of your widget:

   {{<imgproc src="/services/data/visualize-widget-configure.png" alt="Click the pencil icon to configure your widget." style="width:500px" resize="1200x" class="imgzoom fill shadow" >}}

   {{% alert title="Tip" color="tip" %}}
   Select a location and machine for your workspace to populate the widget configuration dropdowns using sensor names and data from that machine.
   {{% /alert %}}

1. To arrange widgets on your workspace, click and drag the grid icon in the top left of your widget:

   {{<imgproc src="/services/data/visualize-widget-move.png" alt="Click the grid icon to move a widget." style="width:500px" resize="1200x" class="imgzoom fill shadow" >}}

For example, the following image shows a time series graph measuring noise over time for a sensor component:

> {{<imgproc src="/services/data/visualize-widget-time-series.png" alt="Time series widget measuring noise over time." style="width:700px" resize="1200x" class="imgzoom fill shadow" >}}

Reviewer: isn't this one redundant with widget types below which shows the time series widget?

Reviewer: Looks like nothing has changed here since last review?

You can mix and match multiple widgets to visualize many kinds of data collected by your machine:

> {{<imgproc src="/services/data/visualize-workspace.png" resize="1000x" style="width: 700px" class="fill imgzoom shadow" declaredimensions=true alt="Workspace containing.">}}

Reviewer: This one I could see being useful for showing what's possible
